### ISSUE TITLE
Issue with YOLOv8 in Docker Container 🐳🕵️‍♂️

### ISSUE THREAD

**Issue by JVPC0D3R** [ultralytics/ultralytics#4907]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Other

### [Q] What's happening:

[A] **First frame gets frozen after tracking starts**

When using the Docker image ultralytics/ultralytics:latest, I created a container to test the YOLOv8 model in that environment. Inside the container, I ran a test code to confirm the YOLOv8 model's functionality in that environment.

```python

from ultralytics import YOLO
import os

os.environ['OPENCV_UI'] = 'GTK'

model = YOLO("yolov8n.pt")

model.track(source=0, stream=False, show=True)


``` 

To make this code work, I configured xhost to allow display, and I also configured OpenCV to use GTK as the graphical interface. When running this code, the first frame of the camera with annotated object detection is displayed, but the image freezes with the following output in the terminal:

```
1/1: 0... Success ✅ (inf frames of shape 640x480 at 30.00 FPS)

WARNING ⚠️ inference results will accumulate in RAM unless `stream=True` is passed, causing potential out-of-memory errors for large sources or long-running streams and videos. See [link](https://docs.ultralytics.com/modes/predict/) for help.

Example:
    results = model(source=..., stream=True)  # generator of Results objects
    for r in results:
        boxes = r.boxes  # Boxes object for bbox outputs
        masks = r.masks  # Masks object for segment masks outputs
        probs = r.probs  # Class probabilities for classification outputs

0: 480x640 1 person, 465.1ms
0: 480x640 1 person, 4.4ms
0: 480x640 1 person, 4.0ms
WARNING ⚠️ Waiting for stream 0
WARNING ⚠️ Waiting for stream 0
WARNING ⚠️ Waiting for stream 0

...

WARNING ⚠️ Waiting for stream 0

```

When encountering this issue:
- ✅ I tried displaying the live video using a simple script with OpenCV, and it worked perfectly. 
- ✅ I also tried a modification of the script where I passed the frame through YOLOv8 ``` result = model(frame) ```, and the predictions worked correctly.
- ✅ In addition I tried to implement [supervision](https://github.com/roboflow/supervision) Bytetrack tracker and worked fine!

Thank you for addressing this issue, and I hope it has been helpful.

### Environment

- OS: Ubuntu 20.04
- Docker Desktop 4.23.0
- image ultralytics/ultralytics:latest (09/14/2023)
- container Python 3.10.11
- ultralytics 8.0.178 (Python package)
- opencv-python 4.8.0
- CUDA Version: 12.0

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@JVPC0D3R I'm pulling a Docker image but in the meantime the code you've provided works correctly for me in a Python 3.11 environment on a local M2 Macbook:

<img width="943" alt="Screenshot 2023-09-15 at 02 00 38" src="https://github.com/ultralytics/ultralytics/assets/26833433/552f8ab5-b68b-4ae4-aa8c-f8f7ccd89874">

**Comment by glenn-jocher** [comment]
@JVPC0D3R also you can construct your own frame by frame tracker (since you said that works correctly above) by following this section in our docs https://docs.ultralytics.com/modes/track/#persisting-tracks-loop

**Comment by glenn-jocher** [comment]
@JVPC0D3R 👋 hi, thanks for letting us know about this possible problem with YOLO 🚀. I'm unable to reproduce your issue as your commands do not by themselves enable webcam access inside Docker images. If I pull our latest image and run your code I see this, which is expected:

<img width="1118" alt="Screenshot 2023-09-15 at 02 09 53" src="https://github.com/ultralytics/ultralytics/assets/26833433/5d54786b-36b1-4b58-b139-c1c491c9685b">

We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub with `git pull` or `pip install -U ultralytics` to ensure your problem has not already been solved in the latest code version.
- ✅ **Unmodified** – Your problem must be reproducible using official Ultrlaytics code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/ultralytics/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by glenn-jocher** [comment]
@JVPC0D3R great news 😃! Your original issue may now be resolved ✅ in PR #4913. I discovered the issue is unrelated to Docker and was related to recent changes in StreamLoader. This fix should be live soon in `ultralytics 8.0.179`. 

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!

**Comment by JVPC0D3R** [comment]
@glenn-jocher 👋 Thank you very much for your responses! I apologize for not providing enough information to reproduce the case. Anyway, this is the command I use to launch my Docker container:

```docker run -it -p554:554 -e DISPLAY=$DISPLAY -v $XSOCK:$XSOCK -e QT_QPA_PLATFORM='xcb' --device /dev/video0 --gpus all --name $CONTAINER_NAME $IMAGE_NAME```

You were right; it was a problem in version 8.0.178. After updating to 8.0.180, I haven't had any issues with the same code. Thank you very much for your help!
